Source/KISAddonConfig.cs

```diff
@@ -22,52 +23,154 @@ sealed class KISAddonConfig : MonoBehaviour {
-          || !avPart.partPrefab || avPart.partPrefab.CrewCapacity < 1) {
-        continue;
+    int loadedPartCount;
+    int loadedPartIndex;
+
+    bool HasPodInventories (Part part)
+    {
+      for (int i = part.Modules.Count; i-- > 0; ) {
+        var inv = part.Modules[i] as ModuleKISInventory;
+        if (inv == null) {
+          continue;
+        }
+        if (inv.invType == ModuleKISInventory.InventoryType.Pod) {
+          return true;
+        }
       }
+      return false;
+    }
 
-      DebugEx.Fine("Found part with CrewCapacity: {0}", avPart.name);
-      for (int i = 0; i < avPart.partPrefab.CrewCapacity; i++) {
-        try {
-          var moduleInventory =
-            avPart.partPrefab.AddModule(typeof(ModuleKISInventory).Name) as ModuleKISInventory;
-          KIS_Shared.AwakePartModule(moduleInventory);
-          var baseFields = new BaseFieldList(moduleInventory);
-          baseFields.Load(nodeSettings.GetNode("EvaInventory"));
-          moduleInventory.podSeat = i;
-          moduleInventory.invType = ModuleKISInventory.InventoryType.Pod;
-          DebugEx.Fine("Pod inventory module(s) for seat {0} loaded successfully", i);
-        } catch {
-          DebugEx.Error("Pod inventory module(s) for seat {0} can't be loaded!", i);
+
+    IEnumerator LoadInventories ()
+    {
+      // Kerbal parts.
+      UpdateEvaPrefab(MaleKerbalEva, nodeSettings);
+      UpdateEvaPrefab(FemaleKerbalEva, nodeSettings);
+
+      // Set inventory module for every pod with crew capacity.
+      DebugEx.Info("Loading pod inventories...");
+      for (loadedPartIndex = 0; loadedPartIndex < loadedPartCount; loadedPartIndex++) {
+        AvailablePart avPart = PartLoader.LoadedPartsList[loadedPartIndex];
+        if (!(avPart.name == MaleKerbalEva || avPart.name == FemaleKerbalEva
+              || avPart.name == RdKerbalEva
+              || !avPart.partPrefab || avPart.partPrefab.CrewCapacity < 1)) {
+          if (HasPodInventories (avPart.partPrefab)) {
+            // pod inventories have already been added to the part by another
+            // mod, so skip the part (probably a dynamic crew capacity)
+			  DebugEx.Info("Skipping " + avPart.partPrefab);
+            continue;
+          }
+          DebugEx.Fine("Found part with CrewCapacity: {0}", avPart.name);
+          AddPodInventories (avPart.partPrefab, avPart.partPrefab.CrewCapacity);
         }
+        yield return null;
+      }
+      done = true;
+    }
+
+    public override bool IsReady ()
+    {
+      return done;
+    }
+
+    public override float ProgressFraction ()
+    {
+      return (float)loadedPartIndex / (float)loadedPartCount;
+    }
+
+    public override string ProgressTitle ()
+    {
+      return "Kerbal Inventory System";
+    }
+
+    public override void StartLoad ()
+    {
+      done = false;
+      loadedPartCount = PartLoader.LoadedPartsList.Count;
+      StartCoroutine (LoadInventories ());
+    }
+  }
+
+  public void Awake() {
```

Comment:
This method will trigger on every scene change. Which may potentially result in an unbounded growing of the loaders list. It should be NO-OP in case of nothing is being loaded. Please, handle it.

Reply:
Because KISAddonConfig is marked to load Instantly and not in any other scenes, it will be loaded only once: right at program start before the loading screen shows.

[KSPAddon(KSPAddon.Startup.Instantly, false)]

Reply:
Ah! Right. I've confused it with the `EveryScene` behavior.